src/hooks/usePreloadAssets.ts

```diff
@@ -0,0 +1,41 @@
+import { useEffect, useRef } from "react";
+
+import { isDefined } from "utils/types";
+
+export interface PreloadAsset {
+  url: string;
+  as?: string;
+  type?: string;
+}
+
+const createLinkElement = ({ url, type, as = "image" }: PreloadAsset) => {
+  const link: HTMLLinkElement = document.createElement("link");
+
+  link.rel = "preload";
+  link.as = as;
+  link.href = url;
+
+  if (isDefined(type)) {
+    link.type = type;
+  }
+
+  return link;
+};
+
+export const usePreloadAssets = (assets: PreloadAsset[]) => {
+  const headRef = useRef<HTMLHeadElement>(document.head);
+
+  return useEffect(() => {
```

Comment:
It just makes `usePreloadAssets()` a de facto `useEffect()`.
Nothing special, since there isn't requirement to manipulate the cleanup callback from the inner hook outside the outer one.

Reply:
removed in 6067f321184ada9bc10ccb2158c32d07389ce3b7 as to not confuse people

Reply:
Since useEffect doesn’t ever return anything to my knowledge, not having the return is correct.